### PR TITLE
Feat: Controller Connector Implementation

### LINKED ISSUE #352: [FEAT] Controller Connector Implementation
## Issue Overview

Add the Controller connector from Cartridge to the UI.

## Proposed Solutions or Ideas

Import the Controller package and modify the README explaining how to run `ngrok` to test the controller successfully.

### PR DESCRIPTION
**Pull request by stephanniegb**
# Feat: Controller Connector Implementation 

Fixes #352 

## Types of change

- [x] Feature
- [ ] Bug
- [ ] Enhancement

### CODE REVIEW

**Comment by stephanniegb** [comment]
Another way to start the server with HTTPS  is by using  `next dev  --experimental-https`. Should i add that to the script?

File: packages/nextjs/services/web3/connectors.tsx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
 import { LAST_CONNECTED_TIME_LOCALSTORAGE_KEY } from "~~/utils/Constants";
+import { controllerInstance } from "~~/utils/scaffold-stark/controller";
 
 const targetNetworks = getTargetNetworks();
 
 export const connectors = getConnectors();
 
 // workaround helper function to properly disconnect with removing local storage (prevent autoconnect infinite loop)
-function withDisconnectWrapper(connector: InjectedConnector) {
+function withDisconnectWrapper(connector: InjectedConnector | Connector) {
```
Comment (metalboyrick) [inline packages/nextjs/services/web3/connectors.tsx:18]: can you check that if you use `Connector` only everything still works? `InjectedConnector` might have been inherited from `Connector`, and if it works, we can remove `InjectedConnector` from here

**Review by metalboyrick — CHANGES_REQUESTED** [review]
Minor comments, will leave the rest to @Nadai2010 and @0xquantum3labs since they have more experience in using Cartridge Controller. 

On that note kindly provide the new package versions on the `README.md`.

File: packages/nextjs/services/web3/connectors.tsx (hunk 1 above)
Comment (stephanniegb) [reply packages/nextjs/services/web3/connectors.tsx:18]: okay

Comment (stephanniegb) [reply packages/nextjs/services/web3/connectors.tsx:18]: It works without it I'll go ahead and remove it

**Comment by stephanniegb** [comment]
> Minor comments, will leave the rest to @Nadai2010 and @0xquantum3labs since they have more experience in using Cartridge Controller.
> 
> On that note kindly provide the new package versions on the `README.md`.

Do you mean I should provide the package of the controller in the  `README.md`.

**Review by Nadai2010 — CHANGES_REQUESTED** [review]
Hello thank you very much for this contribution, it is very good. 

## Devnet:

For now I think it is better to remove controller connection in modal connect, for chains.devnet it should have rpc to slot, and have the local devnet configured with katana.

## Sepolia:

Works perfect

## Mainnet:

Controller asks for the connection well but connects in Sepolia, we see how the balance is incorrect and always shows those of Sepolia network.
Mainnet Balance
Captura desde 2024-12-14 18-37-30

Captura desde 2024-12-14 18-40-26

image

Sepolia Balance
Captura desde 2024-12-14 18-38-58

**Comment by Nadai2010** [comment]
> > Minor comments, will leave the rest to @Nadai2010 and @0xquantum3labs since they have more experience in using Cartridge Controller.
> > On that note kindly provide the new package versions on the `README.md`.
> 
> Do you mean I should provide the package of the controller in the `README.md`?

I think that the version of Cartridge/Controller we are using does not need to be added to the Readme in version specific IMO.

What do you think @jrcarlos2000 ?

**Comment by metalboyrick** [comment]
> > Minor comments, will leave the rest to @Nadai2010 and @0xquantum3labs since they have more experience in using Cartridge Controller.
> > 
> > On that note kindly provide the new package versions on the `README.md`.
> 
> Do you mean I should provide the package of the controller in the  `README.md`?

Ya just indicate the version of the controller package in the readme, thank you!

**Review by Nadai2010 — APPROVED** [review]

**Review by metalboyrick — APPROVED** [review]
looks good on my end

**Pull request merged**